### PR TITLE
fix: React Key Prop

### PR DESCRIPTION
**Pull request by JacobMGEvans**
The error was hidden by the use of shortcut fragments. Added the key prop to prevent edge case issues with React rendering.

### CODE REVIEW

File: packages/wrangler/src/whoami.tsx [hunk 1]
```diff
@@ -65,11 +65,11 @@ function Permissions(props: {
 				</Text>
 				<Text>Scope (Access)</Text>
 				{permissions.map(([type, name]) => (
```
Comment (cameron-robey) [inline packages/wrangler/src/whoami.tsx:67]: Probably better to use index as the key, as `name` will be duplicated
Here name actually is the 'read' / 'write' or sometimes not defined (e.g. for offline_access)
I guess we could use `type` as the key but there could be case where it is duplicated with both read and write so probably safest to use the index

**Review by cameron-robey — CHANGES_REQUESTED** [review]

File: packages/wrangler/src/whoami.tsx (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/whoami.tsx:67]: I'll do `name-index` should prevent any overlap

Comment (cameron-robey) [reply packages/wrangler/src/whoami.tsx:67]: I will admit, when `type` and `name` are not the most descriptive variable names for what they actually are... `scope` and `access` might be better (what we supply as the heading names in the cli output)

File: packages/wrangler/src/whoami.tsx [hunk 2]
```diff
@@ -64,8 +64,8 @@ function Permissions(props: {
 					and re-login.
 				</Text>
 				<Text>Scope (Access)</Text>
-				{permissions.map(([type, name]) => (
-					<Fragment key={name}>
+				{permissions.map(([type, name], index) => (
+					<Fragment key={name + index}>
```
Comment (cameron-robey) [inline packages/wrangler/src/whoami.tsx:68]: Probably best to use `type + index` as `type` is always defined, whereas `name` can be undefined and `undefined + number` can lead us to some NaN issues

**Review by cameron-robey — CHANGES_REQUESTED** [review]

File: packages/wrangler/src/whoami.tsx (hunk 2 above)
Comment (cameron-robey) [reply packages/wrangler/src/whoami.tsx:68]: Or `${type}${index}` would be fine too to avoid the addition

**Review by cameron-robey — APPROVED** [review]

**Pull request merged**